csvcubed/csvcubed/readers/configdeserialiser.py

```diff
@@ -0,0 +1,572 @@
+"""
+Config.json Loader
+__________________
+
+A loader for the config.json.
+"""
+import copy
+import json
+import logging
+
+from typing import Any, Dict, Tuple
+
+from urllib.parse import urlsplit
+
+import pandas as pd
+from jsonschema.exceptions import ValidationError
+from csvcubed.readers.columnschema import *
+
+from csvcubed.inputs import PandasDataTypes, pandas_input_to_columnar_str
+from csvcubed.models.cube import *
+from csvcubed.readers.mapcolumntocomponent import map_column_to_qb_component
+from csvcubed.utils.dict import get_from_dict_ensure_exists, get_with_func_or_none
+from csvcubed.utils.file import read_json_from_file
+from csvcubed.utils.log import start_logging
+from csvcubed.utils.uri import csvw_column_name_safe, uri_safe, read_from_uri
+from csvcubed.utils.validators.schema import validate_dict_against_schema
+from csvcubed.writers.qbwriter import QbWriter
+from csvcubed.utils.qb.validation.cube import validate_qb_component_constraints
+from csvcubed.models.rdf.namespaces import GOV, GDP
+
+
+# Used to determine whether a column name matches accepted conventions
+CONVENTION_NAMES = {
+    "measures": [
+        "measure",
+        "measures",
+        "measures column",
+        "measure column",
+        "measure type",
+        "measure types",
+    ],
+    "observations": [
+        "observation",
+        "observations",
+        "obs",
+        "values",
+        "value",
+        "val",
+        "vals",
+    ],
+    "units": ["unit", "units", "units column", "unit column", "unit type", "unit types"]
+    # 'attribute': ['attribute', 'attributes']  # Not supported for loading cube from convention only.
+}
+
+log = logging.getLogger(__name__)
+start_logging()
+
+
+def _load_resource(resource_path: Path) -> dict:
+    """
+    Load a json schema document from either a File or URI
+    """
+    schema: dict = {}
+
+    if "http" in urlsplit(str(resource_path)).scheme:
+        schema = read_from_uri(resource_path, log)
+
+    else:
+        schema = read_json_from_file(resource_path, log)
+
+    return schema
+
+
+def _get_code_list(
+    column_label: str,
+    maybe_code_list: Optional[Union[bool, str]],
+    info_json_parent_dir: Path,
+    maybe_parent_uri: Optional[str],
+    column_data: PandasDataTypes,
+    maybe_property_value_url: Optional[str],
+) -> Optional[QbCodeList]:
+
+    is_date_time_code_list = (
+        (
+            maybe_code_list is None
+            or (isinstance(maybe_code_list, bool) and maybe_code_list)
+        )
+        and maybe_parent_uri
+        == "http://purl.org/linked-data/sdmx/2009/dimension#refPeriod"
```

Comment:
Cool! Above and beyond the ticket.

Reply:
Recent changes have removed this code as it belonged to the older infojson was of creating codelists